Sources/MongoSwift/SDAM.swift

```diff
@@ -3,12 +3,21 @@ import Foundation
 
 /// A struct representing a network address, consisting of a host and port.
 public struct ServerAddress: Equatable {
+    public enum HostType: String {
+        case ipv4
+        case ip_literal
+        case hostname
+        case unix
+    }
+
     /// The hostname or IP address.
     public let host: String
 
     /// The port number.
     public let port: UInt16
 
+    public var type: HostType?
```

Comment:
rather than having this be optional, can we just assign everything that isn't an IP literal to `.hostname`?